addons/web/static/src/js/model.js

```diff
@@ -403,62 +388,21 @@ odoo.define("web/static/src/js/model.js", function (require) {
             return results;
         }
 
-        /**
-         * Used to assign each state section to its respective extension.
-         * If a model finds a related state in the imported state, it is
-         * imported as its new state. If not, prepareState is called to prepare
-         * a new state.
-         * @param {Object} [importedState={}]
-         */
-        importState(importedState = {}) {
-            const state = Object.assign({}, this.externalState, importedState);
-            for (const extension of this.extensions) {
-                extension.importState(state[extension.name]);
-            }
-        }
-
         /**
          * Must be called after construction and state preparation/import.
          * Waits for all asynchronous work needed by the model extensions to be
          * ready.
          * @returns {Promise}
          */
         async load() {
-            const toLoad = new Map();
-            const loaded = [];
-            // Determine which extensions need to be loaded
-            for (const extension of this.extensions) {
-                const { shouldLoad, dependencies } = extension.getLoadingState();
-                if (shouldLoad) {
-                    for (const dep of dependencies) {
-                        if (!this.extensions.some((e) => e.name === dep)) {
-                            // Any dependency not found in the registered
-                            // extensions is considered as loaded.
-                            loaded.push(dep);
-                        }
-                    }
-                    // Extension will be loaded
-                    toLoad.set(extension.name, extension);
-                } else {
-                    // Extension will not be loaded
-                    loaded.push(extension.name);
-                }
-            }
-            // Iterate on each group of extensions ready to load
-            while (toLoad.size) {
-                const loading = new Map();
-                for (const [name, extension] of toLoad) {
-                    const { dependencies } = extension.getLoadingState();
-                    // A extension can be loaded iff each of its dependency
-                    // is already loaded.
-                    if (dependencies.every((dep) => loaded.includes(dep))) {
-                        loading.set(name, extension.load());
-                        toLoad.delete(name);
+            for (let layer = 0; layer < this.extensions.length; layer++) {
+                const layerPromises = [];
+                for (const extension of this.extensions[layer]) {
+                    if (extension.shouldLoad()) {
+                        layerPromises.push(extension.load());
                     }
                 }
-                // Wait for the current group to be loaded
-                await Promise.all(loading.values());
-                loaded.push(...loading.keys());
+                await Promise.all(layerPromises);
```

Comment:
this does not change the problem I have mentioned before

Reply:
Done